core/integration/features/orders/market-orders-all-market-types.feature

```diff
@@ -0,0 +1,114 @@
+    # GFA market orders should not be accepted during an auction but are accepted here because the 
+    # validation is performed on commands at a stage before we insert orders using the feature test framework.
+   When the parties place the following orders:
+      | party | market id | side | volume | price | resulting trades | type        | tif     | only   | error |
+      | party1| ETH/DEC23 | buy  | 1      | 0     | 0                | TYPE_MARKET | TIF_IOC |        | ioc order received during auction trading |
+      | party1| ETH/DEC23 | buy  | 1      | 0     | 0                | TYPE_MARKET | TIF_FOK |        | fok order received during auction trading |
+      | party1| ETH/DEC23 | buy  | 1      | 0     | 0                | TYPE_MARKET | TIF_GFN |        | gfn order received during auction trading |
+      | party1| ETH/DEC23 | buy  | 1      | 0     | 0                | TYPE_MARKET | TIF_GFA |        |  |
+      | party1| ETH/DEC23 | buy  | 1      | 0     | 0                | TYPE_MARKET | TIF_GTC |        | OrderError: Invalid Persistence |
+      | party1| ETH/DEC23 | sell | 1      | 0     | 0                | TYPE_MARKET | TIF_IOC |        | ioc order received during auction trading |
+      | party1| ETH/DEC23 | sell | 1      | 0     | 0                | TYPE_MARKET | TIF_FOK |        | fok order received during auction trading |
+      | party1| ETH/DEC23 | sell | 1      | 0     | 0                | TYPE_MARKET | TIF_GFN |        | gfn order received during auction trading |
+      | party1| ETH/DEC23 | sell | 1      | 0     | 0                | TYPE_MARKET | TIF_GTC |        | OrderError: Invalid Persistence |
+      | party1| ETH/DEC23 | sell | 1      | 0     | 0                | TYPE_MARKET | TIF_GFA |        |  |
+
+   When the parties place the following orders:
+      | party | market id | side | volume | price | resulting trades | type        | tif     | expires in | only   | error |
+      | party1| ETH/DEC23 | buy  | 1      | 0     | 0                | TYPE_MARKET | TIF_GTT | 50         |        | OrderError: Invalid Expiration Datetime |
+      | party1| ETH/DEC23 | sell | 1      | 0     | 0                | TYPE_MARKET | TIF_GTT | 50         |        | OrderError: Invalid Expiration Datetime |
+
+
+    # place auxiliary orders so we always have best bid and best offer as to not trigger the liquidity auction
+    When the parties place the following orders:
+      | party  | market id | side | volume | price | resulting trades | type       | tif     |
+      | party3 | ETH/DEC23 | buy  | 100    | 900   | 0                | TYPE_LIMIT | TIF_GTC |
+      | party4 | ETH/DEC23 | sell | 100    | 1100  | 0                | TYPE_LIMIT | TIF_GTC |
+      | party3 | ETH/DEC23 | buy  | 1      | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
+      | party4 | ETH/DEC23 | sell | 1      | 1000  | 0                | TYPE_LIMIT | TIF_GTC |
+
+    Then the opening auction period ends for market "ETH/DEC23"
```

Comment:
Same: nit pick about it being better to move forward by N blocks, and maybe adding ticks to placing orders to ensure the behaviour is consistent